### DISCUSSION TITLE
Running Garnet in Cluster Mode

### DISCUSSION THREAD

**Discussion by sarathks2103** [microsoft/garnet#325] [Q&A] [answered]
Team,

While i am running garnet docker in cluser mode i am getting following error 
sarath.sasiairindia.com@Saraths-MacBook-Pro garnet-cs % docker run --privileged garnet-server  --cluster=true  --port 7000
    _________
   /_||___||_\      Garnet 1.0.5 64 bit; cluster mode
   '. \   / .'      Port: 7000
     '.\ /.'        https://aka.ms/GetGarnet
       '.'

Unable to initialize server due to exception: Access to the path '/app/cluster' is denied.

please help me with right command

Regards
Sarath K S

**Comment by akoken** [comment, marked as answer]
@sarathks2103 

Docker images typically run as non-root by default. Looks like write permission needs to be given to the `cluster` directory in Dockerfile. As a workaround, consider setting the root user using the --user flag in your docker command, as demonstrated below.

```bash
docker run --name garnet -p 6379:6379 --ulimit memlock=-1 --user root ghcr.io/microsoft/garnet --cluster true
```

**Reply by badrishc** [reply]
Would appreciate a contribution to give write permission to the cluster directory in docker as well.

**Reply by vazois** [reply]
Alternatively, --checkpointdir could be used to specify a path with write permissions (e.g. /tmp)

**Reply by akoken** [reply]
@badrishc Already fixed in #570  couple of months ago.
